src/Microsoft.Data.SqlClient/src/Microsoft/Data/Sql/SqlDataSourceEnumeratorNativeHelper.cs

```diff
@@ -106,6 +107,10 @@ internal static DataTable GetDataSources()
             return ParseServerEnumString(strbldr.ToString());
         }
 
+#if NET6_0_OR_GREATER
+        [UnconditionalSuppressMessage("Trimming", "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+            Justification = "Filter expression uses only primitive types and trimmer safe")]
+#endif
```

Comment:
I would recommend avoiding suppressions on large methods. These are problematic because:
* The suppression applies to the entire method body. It is typically intended to suppress one or two specific calls, but it covers everything. Later on if somebody adds a new code into the method, and introduces a new trim incompatibility, the suppression will still silence that warning as well - and the justification may be wrong at that time -> introducing a trim incompatibility problem.
* It's hard to read - for example I still can't tell exactly where the suppression applies and which primitive types it's talking about.

I would suggest to refactor the code such that the problematic call is isolated into a local function and the suppression is added onto the local function.

I know it makes the code a little less clean, but it's the price we have to pay for the fact that there's no way to add an attribute (or really any kind of metadata) onto a block of IL instructions.